pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -1169,3 +1170,20 @@ func enableAskpassSidecar(sourceType string, auth configsync.AuthType) bool {
 	}
 	return false
 }
+
+func (r *RepoSyncReconciler) getSecretKnownHost(ctx context.Context, repoSync *v1beta1.RepoSync) bool {
```

Comment:
A couple suggestions here:
- The logic is the same for RootSync and RepoSync, so consolidate this to a single method of `reconcilerBase`
- The Secret is already fetched earlier in the reconcile loop during the Secret validation step. I suggest refactoring this so that we only fetch the Secret once

Reply:
Refactored to:
- combined both r*sync to use reconciler base
- only one getSecret call with the temp data stored in reconcilerBase